src/states/node.rs

```diff
@@ -997,34 +1013,57 @@ impl Node {
         let rp_object = ResourceProof::new(target_size, difficulty);
         let mut proof = rp_object.create_proof_data(&seed);
         let leading_zero_bytes = rp_object.create_proof(&mut proof);
+        let elapsed = start.elapsed();
         let parts = proof.into_iter()
             .chunks(MAX_PART_LEN)
             .into_iter()
             .map(|chunk| chunk.collect_vec())
             .collect_vec();
         let part_count = parts.len();
-        for (part_index, part) in parts.into_iter().enumerate() {
-            let direct_message = DirectMessage::ResourceProofResponse {
-                part_index: part_index,
-                part_count: part_count,
-                proof: part,
-                leading_zero_bytes: leading_zero_bytes,
-            };
-            self.send_direct_message(peer_id, direct_message)?
-        }
-        let elapsed = start.elapsed();
-        trace!("{:?} created proof data in {}.{:06} seconds.  Min section size: {}, Target size: \
-                {}, Difficulty: {}, Seed: {:?}",
+        let mut messages = parts.into_iter()
+            .enumerate()
+            .rev()
+            .map(|(part_index, part)| {
+                DirectMessage::ResourceProofResponse {
+                    part_index: part_index,
+                    part_count: part_count,
+                    proof: part,
+                    leading_zero_bytes: leading_zero_bytes,
+                }
+            })
+            .collect_vec();
+        // There will always be at least one part here - safe to unwrap.
+        let first_message = unwrap!(messages.pop());
```

Comment:
Probably not if target size is 0. We need to check that and potentially handle that separately.